packages/vm/src/evm/precompiles/07-ecmul.ts

```diff
@@ -7,7 +7,7 @@ const bn128 = require('rustbn.js')
 export default function (opts: PrecompileInput): ExecResult {
   assert(opts.data)
 
-  const inputData = opts.data
```

Comment:
Does the rustbn package throw if we input a buffer which is not 128 bytes?

Reply:
No, it does not.

The thing here is that only the first 128 bytes are used anyhow for computation. So this is a plain performance optimization. The size of the input value here is not restricted so this gets unnecessarily slow with some larger input data.